src/Symfony/Component/Process/Tests/PhpExecutableFinderTest.php

```diff
@@ -50,12 +50,32 @@ public function testFindArguments()
     public function testNotExitsBinaryFile()
     {
         $f = new PhpExecutableFinder();
-        $phpBinaryEnv = \PHP_BINARY;
-        putenv('PHP_BINARY=/usr/local/php/bin/php-invalid');
 
-        $this->assertFalse($f->find(), '::find() returns false because of not exist file');
-        $this->assertFalse($f->find(false), '::find(false) returns false because of not exist file');
+        $originalPhpBinary = getenv('PHP_BINARY');
 
-        putenv('PHP_BINARY='.$phpBinaryEnv);
+        try {
+            putenv('PHP_BINARY=/usr/local/php/bin/php-invalid');
+
+            $this->assertFalse($f->find(), '::find() returns false because of not exist file');
+            $this->assertFalse($f->find(false), '::find(false) returns false because of not exist file');
+        } finally {
+            putenv('PHP_BINARY='.$originalPhpBinary);
+        }
+    }
+
+    public function testFindWithExecutableDirectory()
```

Comment:
Only the first change is tested (PHP_BINARY env var). The rest is hard to test since we are in cli mode :sweat: